src/pydata_sphinx_theme/theme/pydata_sphinx_theme/sections/sidebar-primary.html

```diff
@@ -1,4 +1,8 @@
 {% block docs_sidebar %}
+{# Set parent_template_name as a way for child templates included by this
+  template to access the template name. This can be used, for example, when
+  generating HTML IDs #}
+{% set parent_template_name = "sidebar-primary" %}
```

Comment:
not clear to me why this is needed; isn't `get_id_generator` and `create_next_id` sufficient to disambiguate?

Reply:
It was to make the HTML markup nicer. 

The "more" dropdown is rendered twice in the source code: once in the header and once in the sidebar. On desktop, the header dropdown is shown whereas the sidebar dropdown is hidden. Vice versa on mobile.

If we pass `pst-more-nav-links` to the id generator in both places, we would get: 

```
pst-more-nav-links
pst-more-nav-links-1
``` 

But I thought the ids looked better if we appended the parent template name rather than sequentially numbering, like so:

```
pst-more-nav-links-in-header
pst-more-nav-links-in-sidebar-primary
```

But maybe, probably this is a case where cleaner code trumps clearer markup.

Reply:
hmm tradeoff is between devs looking at the rendered HTML source vs devs looking at the template HTML source that generates the rendered source... I'd favor making the template simpler / easier to understand. So I that's the same conclusion as "cleaner code > cleaner markup"